msprime/ancestry.py

```diff
@@ -1146,6 +1146,128 @@ def sim_ancestry(
+    simulation are joined using :meth:`tskit.tables.TableCollection.union()`.
+
+    This method executes code from the pyslim tutorial  'Following up with more
+    coalescent simulation'
+
+    ``sample_size`` defaults to the number of samples present in the input tree sequence.
+
+    In general, all parameters  excepting ``samples`` are whatever the defaults
+    of ``msprime.simulate`` are; this includes recombination rate, so that if neither
+    ``recombination_rate`` or a ``recombination_map`` are provided, there will be
+    *no* recombination.
+
+    Note that ``Ne`` defaults to ``1.0``; you may want to set it explicitly
+    using kwargs.
+
+    :type ts: :class`tskit.TreeSequence`
+    :param ts: tree sequence used to continue siumlation, moving forward in time
+    :param float time: The desired period of time to continue simulation in msprime,
+        in units of generations
+    :param int sample_size: Sample size for :meth:`msprime.simulate`,
+        in units of haploid samples,
+    :param dict kwargs: Any other arguments to :meth:`msprime.simulate`.
+    :type nodes: :class`numpy.ndarray`
+    :param continue_nodes: node IDs in ts to match up to the lineages in the continued
+        simulation, defaults to the samples
+    :return: The :class:`tskit.TreeSequence` object resulting from the
+        union of the input tree equence and the continued simulatoin
+    :rtype: :class:`tskit.TreeSequence`
+    """
+    if continue_nodes is None:
+        continue_nodes = ts.samples()
+    if sample_size is None:
+        sample_size = len(continue_nodes)
+
+    new_ts = simulate(sample_size=sample_size, end_time=time, **kwargs)
+
+    new_nodes = np.where(new_ts.tables.nodes.time == time)[0]
+
+    if len(new_nodes) > len(continue_nodes):
+        raise RuntimeError(
+            f"""More lineages present in new simulation samples than in the old.. \
+            Decrease sample_size to try again. \
+            Setting sample_size={len(continue_nodes)} \
+            or leaving it to default will ensure compliance."""
+        )
+
+    if len(new_nodes) == 0:
+        oldest_node_time = max(
+            [
+                new_ts.node(new_ts.first().roots[i]).time
+                for i in range(0, len(new_ts.first().roots), 1)
+            ]
+        )
+        raise RuntimeError(
+            f"""No surviving lineages in new simulation at merge time. \
+            Depth of new simulation is {oldest_node_time} - alter parameters \
+            to extend simulation or reduce the simulation time."""
+        )
+
+    node_map = np.repeat(tskit.NULL, new_ts.num_nodes)
+    node_map[new_nodes] = np.random.choice(
+        continue_nodes, len(new_nodes), replace=False
+    )
+
+    tables = _adjust_tables_time(ts, time)
+    tables.union(
+        new_ts.tables, node_map, add_populations=False, check_shared_equality=False
+    )
+
+    return tables.tree_sequence()
+
+
+def _adjust_tables_time(ts, time):
```

Comment:
two suggestions:
- let's just move this in to `continue_simulation`, without defining a new method, with a node saying that we can replace it with a call to the tskit method, when available
- this should act on `tables` (which are edited in place), rather than on a tree sequence